plane/src/database/backend_key.rs

```diff
@@ -16,40 +21,63 @@ impl<'a> KeysDatabase<'a> {
         Self { pool }
     }
 
+    pub async fn prevent_renew(&self, backend: &BackendName) -> Result<bool, sqlx::Error> {
+        let result = sqlx::query!(
+            r#"
+            update backend_key
+            set allow_renew = false
+            where
+                id = $1 and
+                allow_renew = true
+            "#,
+            backend.to_string(),
+        )
+        .execute(self.pool)
+        .await?;
+
+        Ok(result.rows_affected() > 0)
+    }
+
     /// Remove a key, ensuring that it is still expired.
     /// Returns Ok(true) if the key was successfully removed.
-    pub async fn remove_key(&self, id: BackendKeyId) -> Result<bool, sqlx::Error> {
+    pub async fn remove_key(&self, backend: BackendName) -> Result<bool, sqlx::Error> {
         let result = sqlx::query!(
             r#"
             delete from backend_key
             where id = $1
-            and last_renewed - now() > $2
+            and expires_at > now()
```

Comment:
Maybe a dumb question: why do we want to prevent deleting keys that have expired?

Reply:
not a dumb question; it's a bug! Should be the other way around.